CapstoneProject/Models/MoodDetection.m

```diff
@@ -15,50 +15,45 @@ @interface MoodDetection ()
 
 @implementation MoodDetection
 
-- (void) initializer{
-    
-    NSArray *sadKeyWords = @[@"sad", @"down", @"terrible", @"depressed"];
-    NSArray *boredKeyWords = @[@"bored", @"ok", @"tired", @"sleepy"];
-    NSArray *happyKeyWords = @[@"happy", @"great", @"esstatic", @"excited"];
-    NSArray *angryKeyWords = @[@"angry", @"mad", @"upset"];
-    
-    
-    self.stopWordsForMoods = [NSMutableDictionary dictionaryWithObjectsAndKeys:
-                                  sadKeyWords,    @"sad",
-                              boredKeyWords,  @"bored",
-                              happyKeyWords, @"happy",
-                              angryKeyWords, @"angry",
-                              nil
-    ];
+- (instancetype)init {
+    if (self = [super init]) {
+        NSArray *sadKeyWords = @[@"sad", @"down", @"terrible", @"depressed"];
+        NSArray *boredKeyWords = @[@"bored", @"ok", @"tired", @"sleepy"];
+        NSArray *happyKeyWords = @[@"happy", @"great", @"esstatic", @"excited"];
+        NSArray *angryKeyWords = @[@"angry", @"mad", @"upset"];
+        
+        
+        self.stopWordsForMoods =  @{sadKeyWords: @"sad", boredKeyWords: @"bored", happyKeyWords: @"happy", angryKeyWords: @"angry"};
+        
+        self.moodCount = [NSMutableDictionary dictionaryWithObjectsAndKeys:
+                          @0, @"sad",
+                          @0, @"bored",
+                          @0, @"happy",
+                          @0, @"angry",
+                          nil];
+    }
     
-    self.moodCount = [NSMutableDictionary dictionaryWithObjectsAndKeys:
-                      [NSNumber numberWithInt:0], @"sad",
-                      [NSNumber numberWithInt:0], @"bored",
-                      [NSNumber numberWithInt:0], @"happy",
-                      [NSNumber numberWithInt:0], @"angry",
-                      nil];
+    return self;
 }
 
-- (NSString *) detectMood:(NSString *) userText{
-    
-    [self initializer];
+- (NSString *)detectMood:(NSString *)userText{
     
-    NSArray * userTextWords = [userText componentsSeparatedByString:@" "];
+    NSArray *userTextWords = [userText componentsSeparatedByString:@" "];
     
     // counts how many times a key word/ associated word comes up
     
-    for(int i = 0; i < userTextWords.count; i++){
-        if([[userTextWords objectAtIndex:i] isEqual:[_stopWordsForMoods objectForKey:@"sad"]]) {
-            NSNumber *sadCountIncremented = [NSNumber numberWithInt:[[_moodCount objectForKey:@"sad"] intValue] + 1];
+    for (NSString *userTextWord in userTextWords) {
+        if ([userTextWord isEqual:_stopWordsForMoods[@"sad"]]) {
```

Comment:
Use `isEqualToString:`, otherwise, it only compares their address, which can't be different even though they are the same word.